components/sensor/as5600.rst

```diff
@@ -0,0 +1,307 @@
+AS5600 12-Bit Magnetic Position Sensor
+======================================
+
+.. seo::
+    :description: Instructions for setting up AS5600 magnetic position sensor / encoder.
+    :image: as5600.jpg
+    :keywords: AS5600 AS5600L
+
+.. _as5600-component:
+
+Component/Hub
+-------------
+
+The ``as5600`` sensor platform you to use your AS5600 (`datasheet <https://ams.com/documents/20143/36005/AS5600_DS000365_5-00.pdf/649ee61c-8f9a-20df-9e10-43173a3eb323>`__,
+`AMS <https://ams.com/en/as5600>`__) or AS5600L (`datasheet <https://ams.com/documents/20143/36005/AS5600L_DS000545_3-00.pdf/7ade6878-7a32-2294-b88d-479d50fab6de>`__,
+`AMS <https://ams.com/en/as5600l>`__) 12-bit magnetic position sensor with ESPHome. Individual sensors will be added
+using the :ref:`AS5600 Sensor Platform <as5600-sensor>`. To use this hub, first setup
+the :ref:`I²C Bus <i2c>` and connect the sensor to the pins specified there.
+
+.. figure:: images/as5600-full.jpg
+    :align: center
+    :width: 50.0%
+
+    AS5600 12-Bit Magnetic Position Sensor.
+
+.. _AMS_AS5600: https://ams.com/en/as5600
+
+.. _AMS_AS5600L: https://ams.com/en/as5600l
+
+.. code-block:: yaml
+
+    as5600:
+      dir_pin: 22
+      start_position: 0deg
+      range: 180deg
+
+Configuration variables:
+************************
+
+- **dir_pin** (*Optional*, int): The pin connected to the AS5600's direction pin.
+  See :ref:`Direction <as5600_direction>` for more information.
+- **direction** (*Optional*, string): The direction that the magnet should rotate to increase values. 
+  Used in combination with the **dir_pin**.
+
+  - ``clockwise`` (default)
+  - ``counterclockwise``
+- **start_position** (*Optional*, int): The raw position that should be considered the start (i.e. ``0``).
+  See :ref:`Position / Range <as5600_position_range>`.
+- **end_position** (*Optional*, int): The raw position that should be considered the end (e.g. 180deg)
+  of the allowable rotation range. Mutually exclusive with **range**. See :ref:`Position / Range <as5600_position_range>`.
+- **range** (*Optional*, int): The allowable rotation range from the **start_position**. Mutually
+  exclusive with **range**. See :ref:`Position / Range <as5600_position_range>`.
```

Comment:
```suggestion
  exclusive with **end_position**. See :ref:`Position / Range <as5600_position_range>`.
```

Reply:
Fixed! Thank you for spotting that typo!